karma.config.js

```diff
@@ -1,8 +1,22 @@
+const globby = require('globby');
+
 process.env.CHROME_BIN = require('puppeteer').executablePath();
 
+let testFilesGlob = process.argv[process.argv.length - 1]; // last argv
```

Comment:
you don't need all the patterns below: 
```test/**/*${testFilesGlob}*``` satisfies all your scenarios

Reply:
Yeah, but that would be too greedy, I want to match only test files and not any file, like in the `preprocessors` config.